build/patches/Enable-crash-reporting-ui.patch

```diff
@@ -3,18 +3,49 @@ Date: Fri, 23 Apr 2021 18:41:48 +0000
 Subject: Enable crash report ui
 
 ---
- .../crash_upload_list_android.cc              |  8 +++-
- chrome/browser/ui/webui/crashes_ui.cc         |  8 +---
- .../crash/core/browser/resources/crashes.js   |  2 +-
- components/crash_strings.grdp                 | 16 +++----
- .../MinidumpUploadCallable.java               | 20 +--------
- .../minidump_uploader/MinidumpUploader.java   | 43 ++++++++-----------
- 6 files changed, 36 insertions(+), 61 deletions(-)
+ .../crash_upload_list_android.cc              |  15 ++-
+ chrome/browser/net/chrome_network_delegate.cc |   7 ++
+ chrome/browser/ui/BUILD.gn                    |   1 +
+ chrome/browser/ui/webui/crashes_ui.cc         | 106 ++++++++++++++++--
+ .../crash/core/browser/resources/crashes.js   |  16 +--
+ components/crash_strings.grdp                 |  16 +--
+ .../minidump_uploader/CrashFileManager.java   |   9 +-
+ .../MinidumpUploadCallable.java               |  20 +---
+ .../minidump_uploader/MinidumpUploader.java   |  29 +----
+ .../upload_list/text_log_upload_list.cc       |   3 +-
+ components/upload_list/upload_list.cc         |  12 +-
+ components/upload_list/upload_list.h          |   6 +
+ 12 files changed, 158 insertions(+), 82 deletions(-)
 
 diff --git a/chrome/browser/crash_upload_list/crash_upload_list_android.cc b/chrome/browser/crash_upload_list/crash_upload_list_android.cc
 --- a/chrome/browser/crash_upload_list/crash_upload_list_android.cc
 +++ b/chrome/browser/crash_upload_list/crash_upload_list_android.cc
-@@ -136,8 +136,14 @@ void CrashUploadListAndroid::LoadUnsuccessfulUploadList(
+@@ -81,11 +81,15 @@ void CrashUploadListAndroid::LoadUnsuccessfulUploadList(
+   const char pending_uploads[] = ".dmp";
+   const char skipped_uploads[] = ".skipped";
+   const char manually_forced_uploads[] = ".forced";
++  const char zipped_uploads[] = ".zip";
+ 
+   base::FileEnumerator files(upload_log_path().DirName(), false,
+                              base::FileEnumerator::FILES);
+   for (base::FilePath file = files.Next(); !file.empty(); file = files.Next()) {
+     UploadList::UploadInfo::State upload_state;
++    if (file.value().find(zipped_uploads) != std::string::npos) {
```

Comment:
In which cases the files would not be `.zip`?

Reply:
I guess those would be the `.dmp` files?

Reply:
in android the upload status is saved by the extension. this is a real example.

![image](https://user-images.githubusercontent.com/29201891/117530717-fa304500-afde-11eb-9bb2-38d5812e9346.png)

in `/cache/Crash Reports` the files to be uploaded are the files with `* .dmp *` in the form `<guid> .dmp <id> .try <number>`
I produce the .zip file removing all extensions, as in the example "73d35758-26ea-441f-9ab6-97ff7d3c8afa.zip".
uploaded files are listed in uploads.log and take the extension `*.forced*`

Reply:
fixed in https://github.com/bromite/bromite/pull/1105/commits/2eacdb9cfab05dd5fcbee058ce545fdd71253fe8

Reply:
Can you use the C++ equivalend of `endsWith`?

Reply:
fixed in https://github.com/bromite/bromite/pull/1105/commits/de5dacb811773f6d07423cec1e97e4689e85e7bc